### PR TITLE
doc: setup zabbix server/agent

### PR DESCRIPTION
**Pull request by kbushgit**
Signed-off-by: Kostiantyn Bushko <kbushko@intellias.com>

### CODE REVIEW

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 1]
```diff
@@ -0,0 +1,227 @@
+
+=== Table of contents
+* <<Install and setup Zabbix server on Ubuntu 16/18>>
+* <<Build an OTA image for QEMU>>
+* <<Run a QEMU image>>
+* <<Run a QEMU image in the background mode>> 
+* <<Add a template and configure a host for Zabbix server>>
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]: This isn't needed--Antora will automatically add a table of contents in the sidebar.

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 2]
```diff
@@ -0,0 +1,227 @@
+
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
= Device monitoring with Zabbix
```
Pages need a title.

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+=== Table of contents
+* <<Install and setup Zabbix server on Ubuntu 16/18>>
+* <<Build an OTA image for QEMU>>
+* <<Run a QEMU image>>
+* <<Run a QEMU image in the background mode>> 
+* <<Add a template and configure a host for Zabbix server>>
+
+
+=== Install and setup Zabbix server on Ubuntu 16/18
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
== Install and setup Zabbix server on Ubuntu 16/18
```

Heading levels should always be sequential--never skipping from `==` to `====`, for example.

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+.. Finish the installation.
++
+To sign in to Zabbix, use the following default user name and password:
++
+----
+user: Admin
+pass: zabbix
+----
+
+=== Build an OTA image for QEMU
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
== Build an OTA image for QEMU
```

See above.

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+....
+IMAGE_INSTALL_append = " zabbix"
+SOTA_COMM_CONF_ZABBIX_SERVER="10.0.2.2"
+SOTA_COMM_CONF_ZABBIX_SERVERACTIVE="10.0.2.2"
+----
++
+NOTE: The default name of the QEMU image is `qemux86-64`.
+
+
+=== Run a QEMU image
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
== Run a QEMU image
```

See above.

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  ExecStart=<absolute path>/meta-updater/scripts/run-qemu-ota --dir <absolute path to your qemux86-64 build folder>/tmp/deploy/images --uboot-enable=no --host-forward="tcp:0.0.0.0:10555-:10050" --no-gui
+  Restart=always
+  RestartSec=5
+  LimitNOFILE=10000
+  [Install]
+  WantedBy=multi-user.target
+----
+
+
+=== Add a template and configure a host for Zabbix server
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
== Add a template and configure a host for Zabbix server
```

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+==== Import a Zabbix template
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
=== Import a Zabbix template
```

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+. Go to the Zabbix server dashboard (localhost/zabbix).
+. Go to **Configuration** > **Templates**.
+. Click **Import**, and then select **aktualizr-monitoring-zabbix-template.xml**.
+. Select **Crete New / Screen**.
+. Click **Import**.
+
+==== Add a host for monitoring
```
Comment (tkfu) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]:
```suggestion
=== Add a host for monitoring
```

**Comment by kbushgit** [comment]
@tkfu thank you very much for your review, could you provide me with the suggestion in which section this guide can be added in nav.adoc?  Thanks in advance

**Comment by tkfu** [comment]
I pushed one commit that adds this to the navigation, makes some minor corrections, and adds a little bit more explanation. @kbushgit, If you can review my changes and make sure they're acceptable, I'm happy to merge this. (Once the DCO issue is resolved, of course. I'd suggest squashing the commits together, as well.)

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+  LimitNOFILE=10000
+  [Install]
+  WantedBy=multi-user.target
+----
+
+
+== Add a template and configure a host for Zabbix server
+
+=== Import a Zabbix template
+link:{attachmentsdir}/aktualizr-monitoring-zabbix-template.xml[Download Zabbix template]
```
Comment (kbushgit) [inline docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]: @tkfu could you look to this link? it should download *.xml file, it's unclear for me how to correctly set path to attachment folder. Thanks

**Comment by kbushgit** [comment]
> I pushed one commit that adds this to the navigation, makes some minor corrections, and adds a little bit more explanation. @kbushgit, If you can review my changes and make sure they're acceptable, I'm happy to merge this. (Once the DCO issue is resolved, of course. I'd suggest squashing the commits together, as well.)

Thanks, all your changes look fine for me, but I'm going to add one more section before pushing this PR to master

File: docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc (hunk 9 above)
Comment (tkfu) [reply docs/ota-client-guide/modules/ROOT/pages/device-monitoring-with-zabbix.adoc]: The way you've done it is correct, for rendering on the docs portal. To get the link to work on github as well, add the following snippet near the top of the file (but below the headline):

```
[NOTE]
====
We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
====
:attachmentsdir: ../assets/attachments
endif::[]
```

**Comment by patrickvacek** [comment]
Looks quite good to me! Let us know when you're ready for final review.

**Comment by patrickvacek** [comment]
This is still listed as a draft but you recently requested another review from Jon. Is it ready for general review?

**Comment by patrickvacek** [comment]
> I can't validate that the instructions work, but as far as I'm concerned this is almost good to go.

I also haven't reproduced, but it looks good to me.

> I'd like to see some of those junky github-added commits squashed, and I think it makes sense to have this on the current docs branch (so it'd need a rebase onto 2020.2-docs).

I think this is the last thing required. You'll need to change which target branch you want to merge this PR into as well.

**Comment by patrickvacek** [comment]
> > I'd like to see some of those junky github-added commits squashed, and I think it makes sense to have this on the current docs branch (so it'd need a rebase onto 2020.2-docs).
> 
> I think this is the last thing required. You'll need to change which target branch you want to merge this PR into as well.

It looks like the squashing has been taken care of, but I think you should still follow Jon's suggestion of basing this branch and PR on the `2020.2-docs` branch. Once that's done, we can merge this.

**Comment by patrickvacek** [comment]
Looks like you still need to get rid of those extra commits. I guess those are extra things from master that aren't in 2020.2-docs yet.

**Comment by kbushgit** [comment]
> Looks like you still need to get rid of those extra commits. I guess those are extra things from master that aren't in 2020.2-docs yet.

that's because my initial commit was based on the master?

**Comment by patrickvacek** [comment]
> that's because my initial commit was based on the master?

Yes, exactly. You'll want to get rid of those extra commits.

**Comment by kbushgit** [comment]
> > that's because my initial commit was based on the master?
> 
> Yes, exactly. You'll want to get rid of those extra commits.

yes, no problem, so it should be based on 2020.2-docs branch?

**Comment by patrickvacek** [comment]
> so it should be based on 2020.2-docs branch?

Yep!

**Comment by kbushgit** [comment]
> > so it should be based on 2020.2-docs branch?
> 
> Yep!
done

**Comment by patrickvacek** [comment]
Are your new changes now contained in the same single commit?

**Comment by kbushgit** [comment]
> Are your new changes now contained in the same single commit?

yes

File: docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+                </group>
+            </groups>
+            <applications>
+                <application>
+                    <name>Aktualizr/Application</name>
+                </application>
+            </applications>
+            <items>
+                <item>
+                    <name>aktualizr.threads.count</name>
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: Did we lose the proc.cpu.util entries?

Comment (kbushgit) [reply docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: I've removed this key, but I can revert it'll take a couple of mins!

Comment (kbushgit) [reply docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: currently, template contain 3 keys for monitoring
```
aktualizr.rss.memory.usage - memory (aktualizr)
aktualizr.threads.count - threads (aktualizr)
vfs.dev.write.sectors -  disk usage (entire system)
```

Comment (patrickvacek) [reply docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: Ah, cool. What was `proc.cpu.util` for, then? Did you end up finding that `aktualizr.rss.memory.usage` was more useful?

Comment (kbushgit) [reply docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: It was just an experiment with Zabbix keys,  this key stands for CPU utilization by the aktualizr, I just do not want to overload template.
Any keys which are defined in Zabbix can be added really in a couple of mins on-demand as separate PR

Comment (patrickvacek) [reply docs/ota-client-guide/modules/ROOT/assets/attachments/aktualizr-monitoring-zabbix-template.xml]: Cool. Then this looks good to go. Thanks!

**Pull request merged**